client/src/tests/ProductDetails/pdrender.test.js

```diff
@@ -1,40 +1,209 @@
+import QuantitySelector from '../../components/ProductDetails/ProductCartActions/QuantitySelector';
+import Selector from '../../components/ProductDetails/ProductCartActions/Selector';
+
 
 describe('render product details', () => {
 
+  it('should render a generic selector', () => {
+    const options = new Array(5); 
+    const onChange = jest.fn(); 
+    render(<Selector name="a select" id="a select" label="some label" onChange={onChange}>
+      {options.map((_,i) => <option value={i}>{i}</option>)}
+    </Selector>);
+
+    const comboBox = screen.getByRole('combobox'); 
+
+    expect(comboBox).toBeTruthy();
+    expect(screen.getByLabelText('some label')).toBeTruthy(); 
+
+    fireEvent.change(comboBox, {target: {value: 2}}); 
+    expect(onChange).toHaveBeenCalledTimes(1); 
+    
+  })
+
+  it('should render sku select', () => {
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+    render(<SizeSelector inStockSkus={inStockSkus} selectedSku={inStockSkus[0][0]} showLabel={true} />);
+
+    expect(screen.getByRole('combobox')).toBeTruthy();
+    expect(screen.getAllByRole('option').length).toBe(inStockSkus.length + 1);
+    expect(screen.getByLabelText('Please select size')).toBeTruthy(); 
+
+  });
+
+  it('should render quantity select', () => {
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+    const selectedSku = inStockSkus[0]
+    render(<QuantitySelector selectedSku={selectedSku[0]} selectedStyle={selectedStyle} selectedQuantity={undefined} setSelectedQuantity={() => {}}/>)
+    
+    expect(screen.getByRole('combobox')).toBeTruthy();
+    expect(screen.getAllByRole('option').length).toBe(selectedSku[1].quantity + 1);
+    expect(screen.getByLabelText(`select quantity for ${selectedStyle?.name} sku ${selectedSku[0]}`)).toBeTruthy(); 
+
+  })
+
   it('should render cart actions', () => {
-    render(<ProductCartActions />);
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+
+    render(<ProductCartActions selectedStyle={selectedStyle} />);
     expect(screen.getAllByRole('button').length).toBe(2);
-    //TODO: complete tests
+    expect(screen.getAllByRole('combobox').length).toBe(2);
+    const addToCartButton = screen.getByText('Add to Cart');
+    expect(addToCartButton).toBeTruthy(); 
+    //click without sku selected
+    fireEvent.click(addToCartButton); 
+    
+    //select a sku
+    const skuSelect = screen.getByText('Select Size');
+    expect(skuSelect).toBeTruthy(); 
+    fireEvent.change(skuSelect, {target: {value: inStockSkus[0][0] }});
+
+    fireEvent.click(addToCartButton); 
```

Comment:
shouldnt you also have a check here to see that a prompt window was rendered with the proper message?